blockindex/contractstaking/indexer.go

```diff
@@ -162,22 +165,26 @@ func (s *Indexer) PutBlock(ctx context.Context, blk *block.Block) error {
 	}
 
 	// commit the result
-	return s.commit(handler)
+	return s.commit(handler, blk.Height())
 }
 
 // DeleteTipBlock deletes the tip block from indexer
 func (s *Indexer) DeleteTipBlock(context.Context, *block.Block) error {
 	return errors.New("not implemented")
 }
 
-func (s *Indexer) commit(handler *contractStakingEventHandler) error {
-	batch, delta, height := handler.Result()
+func (s *Indexer) commit(handler *contractStakingEventHandler, height uint64) error {
+	batch, delta := handler.Result()
+	// update cache
 	if err := s.cache.Merge(delta); err != nil {
 		s.reloadCache()
 		return err
 	}
-	s.cache.PutHeight(height)
 	s.cache.PutTotalBucketCount(s.cache.TotalBucketCount() + delta.AddedBucketCnt())
+	// update indexer height cache
+	s.height.Store(height)
```

Comment:
move to L192, after the batch is successfully written, then update height

Reply:
fixed